File: preview_server.js
/**
 * Minimal Node.js preview server — no npm packages, stdlib only.
 * Proxies all requests to the FastAPI app on port 8000 so the
 * preview panel shows the real app (not just the static mockup).
 *
 * For full FastAPI functionality run:
 *   .venv/bin/uvicorn main:app --reload --host 0.0.0.0 --port 8000
 */
const http = require('http')

const PORT     = parseInt(process.env.PORT || '8002', 10)
const UPSTREAM = { host: '127.0.0.1', port: 8000 }

const server = http.createServer((req, res) => {
  const options = {
    host: UPSTREAM.host,
    port: UPSTREAM.port,
    path: req.url,
    method: req.method,
    headers: req.headers,
  }

  const proxy = http.request(options, (upRes) => {
    res.writeHead(upRes.statusCode, upRes.headers)
    upRes.pipe(res, { end: true })
  })

  proxy.on('error', () => {
    res.writeHead(502)
    res.end('FastAPI app not running — start it with:\n  .venv/bin/uvicorn main:app --reload --host 0.0.0.0 --port 8000')
  })

  req.pipe(proxy, { end: true })
})

server.listen(PORT, '0.0.0.0', () => {
  console.log(`Preview proxy running on http://0.0.0.0:${PORT} → FastAPI :8000`)
})

File: preview_server.py
"""
Minimal stdlib-only HTTP server for Claude preview_start compatibility.
Serves static HTML so the UI is visible in the preview panel.

For full functionality (database, HTMX, API routes) run:
    .venv/bin/uvicorn main:app --reload --host 0.0.0.0 --port 8000
"""
import http.server
import socketserver
import os
import mimetypes

PORT = int(os.environ.get("PORT", 8002))
BASE_DIR = os.path.dirname(os.path.abspath(__file__))


class Handler(http.server.BaseHTTPRequestHandler):

    def do_GET(self):
        if self.path in ('/', '/index.html'):
            self._serve_file('templates/preview.html', 'text/html')
        elif self.path.startswith('/static/'):
            rel = self.path.lstrip('/')
            self._serve_file(rel, mimetypes.guess_type(rel)[0] or 'text/plain')
        else:
            self.send_response(404)
            self.end_headers()

    def _serve_file(self, rel_path, content_type):
        full = os.path.join(BASE_DIR, rel_path)
        try:
            with open(full, 'rb') as f:
                data = f.read()
            self.send_response(200)
            self.send_header('Content-Type', content_type)
            self.send_header('Content-Length', str(len(data)))
            self.end_headers()
            self.wfile.write(data)
        except FileNotFoundError:
            self.send_response(404)
            self.end_headers()

    def log_message(self, fmt, *args):
        pass  # suppress request logs


os.chdir(BASE_DIR)
with socketserver.TCPServer(('0.0.0.0', PORT), Handler) as httpd:
    httpd.allow_reuse_address = True
    httpd.serve_forever()
